jujugui/static/gui/src/app/components/budget-table/row/row.js

```diff
@@ -270,6 +289,108 @@ YUI.add('budget-table-row', function() {
     },
 
     /**
+      Get the list of terms for the application.
+
+      @method _getTerms
+    */
+    _getTerms: function() {
+      // Get the list of terms for the uncommitted apps.
+      const terms = this._getTermIds();
+      // If there are no terms for this application then we don't need to
+      // display anything.
+      if (!terms || terms.length === 0) {
+        this.setState({terms: [], termsLoading: false});
+        return;
+      }
+      this.setState({termsLoading: true}, () => {
+        terms.forEach(term => {
+          const xhr = this.props.showTerms(term, null, (error, term) => {
+            if (error) {
+              console.error('cannot retrieve terms:', error);
```

Comment:
Maybe surface this error to the user? db.notifications.add? Or at least ad a TODO.